pkg/env/env.go

```diff
@@ -336,26 +336,38 @@ func (e *testEnv) Run(m *testing.M) int {
 	setups := e.getSetupActions()
 	// fail fast on setup, upon err exit
 	var err error
+
+	defer func() {
+		// Recover and see if the panic handler is disabled. If it is disabled, panic and stop the workflow.
+		// Otherwise, log and continue with running the Finish steps of the Test suite
+		rErr := recover()
+		if rErr != nil {
+			if e.cfg.IgnorePanicRecovery() {
+				panic(rErr)
+			}
+			klog.Error("Recovering from panic and running finish actions", rErr)
+		}
+
+		finishes := e.getFinishActions()
+		// attempt to gracefully clean up.
+		// Upon error, log and continue.
+		for _, fin := range finishes {
```

Comment:
What happens if panic occurs in Finish steps ?
Will it recover  again ?
If so, should we check to see if panic happened during Finish step ?